Source/Core/Core/PatchEngine.cpp

```diff
@@ -118,7 +118,8 @@ void LoadPatchSection(const std::string& section, std::vector<Patch>* patches,
       if (line[0] == '$')
       {
         // Take care of the previous code
-        if (!currentPatch.name.empty())
+        if (!currentPatch.name.empty() &&
+            AchievementManager::GetInstance().IsPatchApproved(currentPatch))
```

Comment:
Okay, after trying to trace how exactly codes get loaded and applied, I'm pretty sure this is the wrong place to filter. By doing it here you're also affecting the UI in non-obvious ways, which may lead to Dolphin loading a filtered list of patches and then saving the 'modified' list to disk, deleting all patches that were filtered out. You *probably* want to filter `s_on_frame` here, in between those two statements:

https://github.com/dolphin-emu/dolphin/blob/f49659fbfc931fde6478d3c5d0fc445bfb62d3bf/Source/Core/Core/PatchEngine.cpp#L217-L220